providers/providers.go

```diff
@@ -52,6 +52,14 @@ type Provider struct {
 	Path   string
 }
 
+func httpClient() (*http.Client, error) {
+	proxy, err := config.GetAPIProxy()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not parse proxy URL")
+	}
+	return &http.Client{Transport: &http.Transport{Proxy: http.ProxyURL(proxy)}}, nil
```

Comment:
I think we should use `return ranger.NewHttpClient(ranger.WithProxy(proxy))` to be consistent with other implementations. This would also have the advantage that we have proper timeouts set.